### ISSUE TITLE
false-positive no-negated-condition in a if .. else if .. else condition

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#7627]
**Tell us about your environment**

* **ESLint Version:**

v3.10.2

* **Node Version:**

v7.1.0

* **npm Version:**

v4.0.2

**What parser (default, Babel-ESLint, etc.) are you using?**

Default.

**Please show your full configuration:**

```json
{
  "rules": {
    "no-negated-condition": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```diff
if (typeof advert === 'string') {
  advertElement = AdIntegrator.getElementById(advert);
+ } else if (!minidom.isElement(advert)) {
  advertElement = advert[0];
} else {
  advertElement = advert;
}
```

**What did you expect to happen?**

No error.

There is no reasonable way to write this condition without negating `minidom.isElement(advert)`.

**What actually happened? Please include the actual, raw output from ESLint.**

Got an error:

> no-negated-condition Unexpected negated condition.: /AdBehaviour.js:50

**Comment by not-an-aardvark** [comment]
Maybe I'm misunderstanding something, but couldn't you write it like this instead?

```js
if (typeof advert === 'string') {
  advertElement = AdIntegrator.getElementById(advert);
+ } else if (minidom.isElement(advert)) {
  advertElement = advert;
} else {
  advertElement = advert[0];
}
```

**Comment by gajus** [comment]
> Maybe I'm misunderstanding something, but couldn't you write it like this instead?

Yes, you are right. I think my processor skipped a tick. :-)
